File: markdown/defaults.go
package markdown

type Options struct {
	baseUrl      string
	breaks       bool
	gfm          bool
	headerIds    bool
	headerPrefix string
	langPrefix   string
	highlight    interface{}
	mangle       bool
	pedantic     bool
	renderer     *Options
	sanitize     bool
	sanitizer    interface{}
	silent       bool
	smartLists   bool
	smartypants  bool
	tokenizer    *Tokenizer
	walkTokens   interface{}
	xhtml        bool
}

func GetDefaults() *Options {
	return &Options{
		baseUrl:      "",
		breaks:       false,
		gfm:          true,
		headerIds:    true,
		headerPrefix: "",
		highlight:    "",
		langPrefix:   "language-",
		mangle:       true,
		pedantic:     false,
		sanitize:     false,
		sanitizer:    nil,
		silent:       false,
		smartLists:   false,
		smartypants:  false,
		tokenizer:    nil,
		walkTokens:   nil,
		xhtml:        false,
	}
}

File: lex/src/defaults.js
function getDefaults() {
  return {
    baseUrl: null,
    breaks: false,
    gfm: true,
    headerIds: true,
    mangle: true,

    headerPrefix: '',
    highlight: null,
    langPrefix: 'language-',

    pedantic: false,
    renderer: null,
    sanitize: false,
    sanitizer: null,
    silent: false,
    smartLists: false,
    smartypants: false,
    tokenizer: null,
    walkTokens: null,
    xhtml: false
  };
}

function changeDefaults(newDefaults) {
  module.exports.defaults = newDefaults;
}

module.exports = {
  defaults: getDefaults(),
  getDefaults,
  changeDefaults
};
